### ISSUE TITLE
[BUG] Can't send messages to the Power Automate flow

### ISSUE THREAD

**Issue by savissimo** [prometheus-msteams/prometheus-msteams#351]
**Describe the bug**
I followed instructions and installed the receiver in my OpenShift cluster. However, I got no messages in Teams (yes, there are alerts firing). I find a stream of these logs in my pod:

```
2024/05/30 16:14:14 [DEBUG] POST https://prod-251.westeurope.logic.azure.com:443/workflows/<redacted>/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%2Ftriggers%2Fmanual%2Frun&sv=1.0&sig=<redacted>
{"alert":{"receiver":"Critical Teams","status":"firing","alerts":[{"status":"firing","labels":{"alertname":"Outbound errors queue length","instance":"10.128.2.24:8080","job":"outboundlander-service","namespace":"<redacted>","pod":"outboundlander-66fb7b648d-rwdm8","prometheus":"openshift-user-workload-monitoring/user-workload","service":"outboundlander-service","severity":"critical"},"annotations":{"message":"Ci sono molti messaggi nella coda di errori dell'outbound."},"startsAt":"2024-05-30T14:16:50.105Z","endsAt":"0001-01-01T00:00:00Z","generatorURL":"https://thanos-querier-openshift-monitoring.apps.<redacted>/api/graph?g0.expr=rabbitmq_outbound_error_queue_length%7Bjob%3D%22outboundlander-service%22%2Cnamespace%3D%22<redacted>%22%7D+%3E+500&g0.tab=1","fingerprint":"95bf367cd4110cf4"},{"status":"firing","labels":{"alertname":"Outbound errors queue length","instance":"10.128.2.24:8081","job":"outboundlander-service","namespace":"<redacted>","pod":"outboundlander-66fb7b648d-rwdm8","prometh...
```

Followed by this:

```
{"caller":"logging.go:25","err":null,"level":"debug","response_message":"{\"error\":{\"code\":\"InvalidRequestContent\",\"message\":\"The input body for trigger 'manual' of type 'Request' must be of type JSON, but was of type 'application/octet-stream'.\"}}","response_status":400,"ts":"2024-05-30T16:16:54.991879121Z","webhook_url":"https://prod-251.westeurope.logic.azure.com:443/workflows/<redacted>/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%2Ftriggers%2Fmanual%2Frun&sv=1.0&sig=<redacted>"}
{"caller":"transport.go:66","host":"promteams.prometheus-msteams.svc.cluster.local:2000","method":"POST","status":200,"took":"71.340822ms","ts":"2024-05-30T16:16:54.991951287Z","uri":"/alertmanager"}
```

**Version**
Docker image v.1.5.2

**To Reproduce**
I followed the instructions from here: https://examples.openshift.pub/cluster-configuration/monitoring/alertmanager/microsoft-teams-receiver/

**Expected behavior**
The Power Automate flow accepts the request and posts the message.

**Comment by tekphobia** [comment]
I resolved this with the following:

`+++ b/pkg/service/service.go
@@ -83,6 +83,8 @@ func (s simpleService) post(ctx context.Context, c card.Office365ConnectorCard,
                return pr, err
        }

+       req.Header.Set("Content-Type", "application/json")
+
        resp, err := s.client.Do(req)
        if err != nil {
                err = fmt.Errorf("http client failed: %w", err)`

**Comment by savissimo** [comment]
Great news, thank you! Will there be a Docker release for this fix?

**Comment by simonesaviolo-reda** [comment]
In fact I used the service in conjunction with a Power Automate flow. This may be the reason for the bug.

**Comment by tekphobia** [comment]
> Great news, thank you! Will there be a Docker release for this fix?

I would imagine once the PR is merged you will be able to build the docker image with:
`make docker`

**Comment by savissimo** [comment]
Should I push that image to a private repository of mine or will it be pushed to the project's repository on quay.io?

**Comment by Unyxos** [comment]
Hi :) 
Has anyone been able to use Power Automate flows after changing the line of code to send as `application/json` ?

So far here, I keep getting errors because the format expected by Power Automate is different (seems they only support Adaptive Cards being sent through webhooks ?)

**Comment by atc0005** [comment]
FWIW, here are some additional references that you may find useful:

- https://gist.github.com/chusiang/895f6406fbf9285c58ad0a3ace13d025
  - lots of discussion & an example script to use adaptive cards format
- https://github.com/tlolkema/simple-teams-message
  - the `action.yml` file contains an example payload that could be adapted for use here

**Comment by pbenefice** [comment]
I believe an MS Teams integration was added directly into Alert Manager supporting Flows (not tested yet on my side)
![image](https://github.com/user-attachments/assets/6c45f05d-9c7d-4c62-b9cd-f10671b3b559)

https://github.com/prometheus/alertmanager/pull/4024

**Comment by savissimo** [comment]
> I believe an MS Teams integration was added directly into Alert Manager supporting Flows (not tested yet on my side) 
> 
> [prometheus/alertmanager#4024](https://github.com/prometheus/alertmanager/pull/4024)

That's true, but some users reported that the new integration offers much less flexibility than this service. It would be great to be able to choose what to use.

**Comment by MeesvanStraten** [comment]
I still have this problem with the HELM chart deployment option. When calling the webhook endpoint i get: **The input body for trigger 'manual' of type 'Request' must be of type JSON, but was of type 'application/octet-stream'.\"}}","response_status":400**.
Its the PowerAutomate flow for a channel, i can't find a option in the docs to fix this, is it something I am doing wrong?
